### ISSUE TITLE
Support UAVCAN v1

### ISSUE THREAD

**Issue by pavel-kirienko** [skyyuzhang/UAVCAN_NODE_FreeRTOS#1]
I am noticing that the link to this repository posted on the UAVCAN forum is getting a lot of attention, which implies that the project is helpful to our adopters. It would be even more helpful if it supported [UAVCAN v1](https://forum.uavcan.org/t/uavcan-v1-crash-course/778) instead of v0. Let me know if you have any questions or need assistance with v1. Thank you.

**Comment by thopiekar** [comment]
@pavel-kirienko Looks like the author of the project lost focus to check the issues.
From my personal side, I'm really interested in a practical example to build UAVCAN nodes and would love to have an updated example.
It would be also nice to get some code snippets to quickly get the node accessed via Ardupilot or PX4.
At the moment, when reading the docs there is a lot of theory described and I can't easily follow. Might be trivial for people who put a lot of work into the project, but for me it is difficult.

Thanks! 😉

**Comment by pavel-kirienko** [comment]
We have very hands-on documentation with examples provided for every implementation library:

- https://pyuavcan.readthedocs.io/en/latest/
- https://github.com/UAVCAN/libcanard

These should be trivial to follow for anybody with embedded experience. What we lack is a complete demo that runs on hardware; I thought this project would be a nice solution to that if it were to be updated to support v1.

**Comment by thopiekar** [comment]
I agree. The board mentioned here is one you can get very cheap actually and gives you the needed CAN interface via HAL.
Personally, I miss the glue code (previously placed in drivers/) and from the docs I can't really follow where they went with V1.

Something that would be really nice is if you could take over this project and take it as a first practical reference how to develop nodes. From experience, I learn better with practical examples and I can imagine other people do so, too! :)

**Comment by pavel-kirienko** [comment]
You may find this relevant: https://github.com/UAVCAN/platform_specific_components/issues/11
